File: src/main/java/com/concesionario/entity/Employee.java
package com.concesionario.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "employees")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @Column(name = "employee_code", nullable = false, unique = true, length = 20)
    private String employeeCode;

    @Column(length = 100)
    private String department;

    @Column(name = "hire_date", nullable = false)
    private LocalDate hireDate;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;
}

// =============================================
// ---- Brand.java ----

File: frontend/src/types/employee.ts
export interface EmployeeRequest {
  userId: string
  firstName: string
  lastName: string
  employeeCode: string
  department?: string
  hireDate: string
  isActive?: boolean
}

export interface EmployeeResponse {
  id: string
  userId: string
  userEmail: string
  firstName: string
  lastName: string
  fullName: string
  employeeCode: string
  department: string | null
  hireDate: string
  isActive: boolean
  createdAt: string
  updatedAt: string
}
